provider/azure/internal/ad/client.go

```diff
@@ -31,3 +36,54 @@ func NewManagementClient(baseURI string) ManagementClient {
 		APIVersion: APIVersion,
 	}
 }
+
+// WithOdataErrorUnlessStatusCode returns a RespondDecorator that emits an
+// azure.RequestError by reading the response body unless the response HTTP status code
+// is among the set passed.
+//
+// If there is a chance service may return responses other than the Azure error
+// format and the response cannot be parsed into an error, a decoding error will
+// be returned containing the response body. In any case, the Responder will
+// return an error if the status code is not satisfied.
+//
+// If this Responder returns an error, the response body will be replaced with
+// an in-memory reader, which needs no further closing.
+func WithOdataErrorUnlessStatusCode(codes ...int) autorest.RespondDecorator {
+	return func(r autorest.Responder) autorest.Responder {
+		return autorest.ResponderFunc(func(resp *http.Response) error {
+			err := r.Respond(resp)
+			if err == nil && !autorest.ResponseHasStatusCode(resp, codes...) {
+				var oe odataRequestError
+				defer resp.Body.Close()
```

Comment:
The whole thing is a straight-up clone of what's in go-autorest, with a very small change that is now documented.

Which reminds me, I need to add the go-autorest LICENSE file.